File: 백준 Study/반복문/num25304.py
# 백준 25304번 영수증 문제
X = int(input())
N = int(input())
sum=0
for i in range(N) :
  A, B = map(int, input().split())
  sum+=(A*B)

if X == sum :
    print('Yes')
else :
    print('No')

File: 백준 Study/반복문/num25304.java
import java.util.Scanner;

public class num25304 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int sum = sc.nextInt();
        int tcn = sc.nextInt();
        int[] arr = new int[tcn];
        int recipe_sum=0;
        for(int i=0; i<tcn; i++) {
            int a = sc.nextInt();
            int b=  sc.nextInt();
            arr[i] = a*b;
            recipe_sum+=arr[i];
        }
        if( sum == recipe_sum) {
            System.out.println("Yes");
        }
        else System.out.println("No");
        sc.close();
    }
}